### PR TITLE
Fix bug with date format when importing to inventory [SCI-4450]

### PR DESCRIPTION
**Pull request by mlorb**
Jira ticket: [SCI-4450](https://biosistemika.atlassian.net/browse/SCI-4450)

### What was done
Fix bug with date format when importing to inventory.
Tests were added.

### CODE REVIEW

File: app/utilities/repository_import_parser/importer.rb [hunk 1]
```diff
@@ -137,7 +137,7 @@ def import_batch_to_database(full_row_import_batch)
           cell_value = column.data_type.constantize.import_from_text(
             value,
             cell_value_attributes,
-            user: @user.as_json
```
Comment (okriuchykhin) [inline app/utilities/repository_import_parser/importer.rb]: It should be serialized

File: app/models/repository_date_time_value_base.rb [hunk 2]
```diff
@@ -26,7 +26,7 @@ def update_data!(new_data, user)
   end
 
   def self.import_from_text(text, attributes, options = {})
-    date_format = options.dig(:user, :settings, :date_format) || Constants::DEFAULT_DATE_FORMAT
+    date_format = options[:user].settings[:date_format] || Constants::DEFAULT_DATE_FORMAT
```
Comment (okriuchykhin) [inline app/models/repository_date_time_value_base.rb]: What if :user is missing?

File: app/utilities/repository_import_parser/importer.rb (hunk 1 above)
Comment (radiokills) [reply app/utilities/repository_import_parser/importer.rb]: Can you plz explain why it should be serialized.

Comment (okriuchykhin) [reply app/utilities/repository_import_parser/importer.rb]: In case we use it in delayed jobs, to preserve user settings for that specific point in time

Comment (okriuchykhin) [reply app/utilities/repository_import_parser/importer.rb]: @user.as_json(root: true, only: :settings)

**Comment by mlorb** [comment]
I added another [PR](https://gitlab.com/biosistemika/scinote/addons/audit_trails/-/merge_requests/54)

**Pull request merged**